lib/trento_web/openapi/schema/checks_catalog.ex

```diff
@@ -0,0 +1,178 @@
+defmodule TrentoWeb.OpenApi.Schema.ChecksCatalog do
+  @moduledoc false
+
+  require OpenApiSpex
+  alias OpenApiSpex.Schema
+
+  alias TrentoWeb.OpenApi.Schema.Provider
+
+  defmodule Check do
+    @moduledoc false
+
+    OpenApiSpex.schema(%{
+      title: "Check",
+      description: "An available check to be executed on the target infrastructure",
+      type: :object,
+      properties: %{
+        sap_system_id: %Schema{type: :string, description: "Check ID", format: :uuid},
+        name: %Schema{type: :string, description: "Check Name"},
+        description: %Schema{type: :string, description: "Check Description"},
+        remediation: %Schema{type: :string, description: "Check Remediation"},
+        implementation: %Schema{type: :string, description: "Check Implementation"},
+        labels: %Schema{type: :string, description: "Check Labels"},
+        premium: %Schema{
+          type: :boolean,
+          description: "Indicates whether the current check is a Premium check"
+        },
+        group: %Schema{
+          type: :string,
+          description: "Check Group, available when requiring a Flat Catalog"
+        },
+        provider: Provider.SupportedProviders
```

Comment:
Where is this `SupportedProviders` specified?

Reply:
That's defined in `lib/trento_web/openapi/schema/provider.ex:7`

```elixir
defmodule SupportedProviders do
  @moduledoc false

  OpenApiSpex.schema(%{
    title: "SupportedProviders",
    type: :string,
    description: "Detected Provider where the resource is running",
    enum: [:azure, :aws, :gcp, :unknown]
  })
end
```
And now that I see it I think:
- should we consider `default` part of that list?
- is `:unknown` actually a _SupportedProvider_?

I believe we can iterate more on this details.

Reply:
`:unknown` is not supported in the checks catalog. We have in the code as an incoming provider in hosts, as we will have hosts that are not running in our known providers, and they will have this value.
So, in this `SupportedProviders` I see it totally right.
`default` is not part of this list, as I don't see any provider option as default